Plex mark devices unavailable if they 'vanish' and clear media

ryanm101 opened the pull request:
Partial Fix for:
https://github.com/home-assistant/home-assistant/issues/10074

if a client vanishes it will persist in UI however it is marked Unavailable and media images are cleared

ryanm101 committed:
Marks Devices unavailable if they 'vanish' and clears media

ryanm101 committed:
Fixed PEP8 complaint

ryanm101 committed:
Fixed Linting

ryanm101 committed:
Lint Fix

ryanm101 committed:
Fix redine of id

ryanm101 committed:
More lint fixes

syssi commented on homeassistant/components/media_player/plex.py:
```diff
@@ -259,6 +265,7 @@ def __init__(self, config, device, session, plex_sessions,
         """Initialize the Plex device."""
         self._app_name = ''
         self._device = None
+        self._is_device_available = False
```
Please use `self._available`.

ryanm101 replied:
ok

syssi commented on homeassistant/components/media_player/plex.py:
```diff
@@ -189,6 +192,9 @@ def update_devices():
         if new_plex_clients:
             add_devices_callback(new_plex_clients)
 
+        for cid in hass.data[PLEX_DATA].keys():
```
Why didn't you use this loop: https://github.com/ryanm101/home-assistant/blob/4d8a8606499b42936d7ad0ca9d77cf13b60a725e/homeassistant/components/media_player/plex.py#L187-L190

ryanm101 replied:
yes this could be moved.

syssi commented on homeassistant/components/media_player/plex.py:
```diff
@@ -154,11 +154,14 @@ def update_devices():
             return
 
         new_plex_clients = []
+        available_ids = []
```
I would prefer `available_client_ids` here.

ryanm101 replied:
Yes that would make more sense

syssi commented:
Did you see `self._is_player_available`? What's the difference?

syssi reviewed (changes requested):
Just remove all your code :wink: and use `_is_player_available` as identifier for `def available`. There is a bug somewhere. Please don't implement duplicated logic.

ryanm101 commented:
Removing all my code and using player available results in player IDLE not unavailable

ryanm101 committed:
Removed redundant loop for setting availability of client
Renamed '_is_device_available' to '_available'
Renamed 'available_ids' to 'available_client_ids'

ryanm101 committed:
removed whitespace per houndCI

ryanm101 commented:
@syssi leaving _player_available till next cleanup (i've to add code to remove unavailable clients and update api to 3.0.6). But Have added prev comments

@balloob Can this make 0.65?
